gcloud-java-pubsub/src/main/java/com/google/cloud/pubsub/PubSub.java

```diff
@@ -387,25 +387,107 @@ public static PullOption maxConcurrentCallbacks(int maxConcurrency) {
 
   MessageConsumer pullAsync(String subscription, MessageProcessor callback, PullOption... options);
 
+  /**
+   * Acknowledges the given messages for the provided subscription. Ack ids identify the messages to
+   * acknowledge, as returned in {@link ReceivedMessage#ackId()} by {@link #pull(String, int)} and
+   * {@link #pullAsync(String, int)}.
+   *
+   * @param subscription the subscription whose messages must be acknowledged
```

Comment:
I believe we are quite close to what Guava does. See [Lists.asList](https://github.com/google/guava/blob/master/guava/src/com/google/common/collect/Lists.java#L312) or other methods.